connect/transforms/src/test/java/org/apache/kafka/connect/transforms/CastTest.java

```diff
@@ -320,6 +321,40 @@ public void castWholeRecordValueSchemalessUnsupportedType() {
         xformValue.apply(new SourceRecord(null, null, "topic", 0, null, Collections.singletonList("foo")));
     }
 
+    @Test
+    public void castLogicalToPrimitive() {
+        Date day = new Date(MILLIS_PER_DAY);
+        xformValue.configure(Collections.singletonMap(Cast.SPEC_CONFIG,
+            "date:int32,timestamp:int64,time:int64,bigdecimal:int64"));
+
+        SchemaBuilder builder = SchemaBuilder.struct();
+        builder.field("date", org.apache.kafka.connect.data.Date.SCHEMA);
+        builder.field("timestamp", Timestamp.SCHEMA);
+        builder.field("time", Time.SCHEMA);
+        builder.field("bigdecimal", Decimal.schema(new BigDecimal(42).scale()));
+        Schema supportedTypesSchema = builder.build();
+
+        Struct recordValue = new Struct(supportedTypesSchema);
+        recordValue.put("date", day);
+        recordValue.put("timestamp", new Date(0));
+        recordValue.put("time", new Date(1));
+        recordValue.put("bigdecimal", new BigDecimal(42));
+
+        SourceRecord transformed = xformValue.apply(
+            new SourceRecord(null, null, "topic", 0,
+                supportedTypesSchema, recordValue));
+
+        assertEquals(1, ((Struct) transformed.value()).get("date"));
+        assertEquals(0L, ((Struct) transformed.value()).get("timestamp"));
+        assertEquals(1L, ((Struct) transformed.value()).get("time"));
+        assertEquals(42L, ((Struct) transformed.value()).get("bigdecimal"));
+
+        Schema transformedSchema = ((Struct) transformed.value()).schema();
+        assertEquals(Schema.INT32_SCHEMA.type(), transformedSchema.field("date").schema().type());
```

Comment:
Could also do the following to be a bit more succinct:
```suggestion
        assertEquals(Schema.Type.INT32, transformedSchema.field("date").schema().type());
```